lib/site_prism/element_container.rb

```diff
@@ -69,6 +69,18 @@ def raise_if_block(obj, name, has_block)
 
     private
 
+    def combine_args(arg_array1, arg_array2)
+      args = []
+      args += arg_array1
+      options1 = (args.pop if args[-1].is_a?(Hash)) || {}
```

Comment:
maybe we should refactor the arguments to their own class?  I wouldn't say it's necessary for this PR, that's up to @natritmeyer but it seems like this method and the logic within is not the responsibility of an ElementContainer.  It is probably worth a refactor at some point.